src/blocks/weather.rs

```diff
@@ -224,11 +293,28 @@ pub async fn run(config: &Config, api: &CommonApi) -> Result<()> {
 
         select! {
             _ = sleep(config.interval.0) => (),
-            _ = api.wait_for_update_request() => ()
+            _ = api.wait_for_update_request() => (),
+            Some(action) = actions.recv() => match action.as_ref() {
+                    "toggle_format" => {
+                        if let Some(ref mut format_alt) = format_alt {
+                            std::mem::swap(format_alt, &mut format);
+                        }
+                    }
+                    _ => (),
+                }
```

Comment:
I'm not sure if fetching the weather on each `toggle_foramt` is a good idea. It consumes API calls and most importantly makes the block less responsive (there is a noticeable delay before the block changes its format).